Assets/Scripts/Entity/Player/Player.cs

```diff
@@ -58,17 +70,58 @@ public void FireWeapon()
         }
     }
 
-
+    
     /* Called when this Player encounters another object.
      */
-    private void OnTriggerEnter(Collider other)
+    private void OnCollisionEnter(Collision other)
     {
-        // Detecting projectiles
-        if (other.gameObject.CompareTag("EnemyProjectile") && Health != 0)
+        // Don't bother doing anything if we can't take damage anyway
+        if(!invincible)
         {
-            Projectile projectile = other.gameObject.GetComponent<Projectile>();
-            ChangeHealthBy(projectile.damage);
-            Destroy(other.gameObject);
+            // Detecting projectiles
+            if (other.gameObject.CompareTag("EnemyProjectile"))
+            {
+                Projectile projectile = other.gameObject.GetComponent<Projectile>();
+                ChangeHealthBy(projectile.damage);
+                Destroy(other.gameObject);
+            }
+
+            // Collision with enemy body
+            else if(other.gameObject.CompareTag("Enemy"))
+            {
+                ChangeHealthBy(-1);
+            }
+        }        
+    }
+
+
+    /* Called every frame for which a collision persists.
+     */ 
+    private void OnCollisionStay(Collision other)
+    {
+        if(!invincible && other.gameObject.CompareTag("Enemy"))
+        {
+            ChangeHealthBy(-1);
+        }
+    }
+
+
+    /* Temporarily turns the player invincible for invincibilityDurationSeconds seconds.
+     */
+    private IEnumerator BecomeTemporarilyInvincible()
+    {
+        Debug.Log("Player became invulnerable to damage!");
+        invincible = true;
+        float tick = 1 / invincibilityDurationSeconds;
+
+        for(float i = 0; i < invincibilityDurationSeconds; i += tick)
+        {
+            playerRenderer.enabled = !playerRenderer.enabled;
+            yield return new WaitForSeconds(tick);
```

Comment:
This logic causes a (perhaps unintended) behavior where the longer the invincibility duration, the fast the player will "flash". Also the slower the duration, slower the player will flash. Like if invincibilityDurationSeconds is 1, the player will become invisible, then visible again after a second. Vs 10, where the player will flicker ever 1/10th of a second, 100 times.

Reply:
Good point, I should've made a constant independent of the duration. Do you like the current rate? I feel like it should be just _slightly_ faster.

Reply:
Maybe faster, if anything maybe make the time spent invisible shorter. That way the player doesn't lose their character. (Alternatively make the player translucent but that may be more difficult)